accounts.go

```diff
@@ -3,14 +3,48 @@ package main
 import (
 	"sync"
 
+	"time"
+
 	"github.com/distributeddesigns/currency"
 )
 
+type portfolio map[string]uint
+
 // Account : State of a particular Account
 type account struct {
 	userID  string
 	balance currency.Currency
 	sync.Mutex
+	pendingBuys    buyStack
+	stockPortfolio portfolio
+}
+
+func cleanBuyStacks() {
+	consoleLog.Debug("Buy Cleanup is Running")
+	for 0 < 1 {
+		for _, account := range accountStore {
+			account.Lock()
+			oldestBuy, err := account.pendingBuys.headPeek()
+			account.Unlock()
+			if err != nil {
+				continue
+			}
+			for oldestBuy.isExpired() {
+				consoleLog.Debugf("Pending buy for %s expired, auto cancelling", account.userID)
+				account.Lock()
+				account.pendingBuys.dequeue()
+				refund := oldestBuy.amount
+				oldestBuy, err = account.pendingBuys.headPeek()
+				account.Unlock()
+				account.AddFunds(refund)
+				if err != nil {
+					break
+				}
+			}
+		}
+		time.Sleep(time.Second * time.Duration(config.CleanPolicy.CleanFrequency))
```

Comment:
The total length of a cleaning cycle will be `time to clean + time sleeping > time sleeping`. Your cycle will drift away from executing every `CleanFrequency` in user time. 

Since the logic behind choosing a `CleanFrequency` is tied to the business logic it needs to be decoupled from the variable `time to clean`.

You can use a [ticker](https://gobyexample.com/tickers) to create an event that fires at a fixed interval.